### ISSUE TITLE
Add ignore: ["inside-function"] to declaration-property-unit-allowed-list

### ISSUE THREAD

**Issue by abea** [stylelint/stylelint#5140]
> Clearly describe the bug

`declaration-property-unit-allowed-list` trips on SCSS function contents. Specifically color functions, which it should ignore.

> Which rule, if any, is the bug related to?

`declaration-property-unit-allowed-list`

> What code is needed to reproduce the bug?

```scss
$input-border: #639;

button {
  border: 1px solid lighten($input-border, 20%);
}
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "declaration-property-unit-allowed-list": {
      "/^border/": ["px"]
    },
  }
}
```

> Which version of stylelint are you using?

`^13.8.0`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

VS Code stylelint plugin

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, it's related to SCSS functions

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors did you get)?

e.g. "The following warnings were flagged:"

```
Unexpected unit "%" for property "border" (declaration-property-unit-allowed-list) stylelint declaration-property-unit-allowed-list
```

**Comment by jeddy3** [comment]
@abea Thanks for the report and for using the template.

I can reproduce this using vanilla CSS on [the demo](https://stylelint.io/demo#N4Igxg9gJgpiBcIBGBXALmiA7ABMAOrjkhAE6ynw4CMADgB44DOEANgJZQ4AWTrAFACYADLADmOYQFIcImQHocADmkBKANyEAvoRAAaEEwCeWNAEN6CcEyb7w2AGbsxVgkXwhSKVjCYeqbjhBOB6wYKxmpGZo7NgAtLSkELQwpGhGcShY7GhxZqysEADuMFBxHExo-niEwcEe8gB6JOSp8tUA2h4MHgC6tcE6RENaIFpAA):

```css
button {
  border: 1px solid hsl(20deg 0% 20% / 80%);
}
```

Gives:

```
2:25 error Unexpected unit "deg" for property "border" (declaration-property-unit-allowed-list)
2:31 error Unexpected unit "%" for property "border" (declaration-property-unit-allowed-list)
2:34 error Unexpected unit "%" for property "border" (declaration-property-unit-allowed-list)
2:40 error Unexpected unit "%" for property "border" (declaration-property-unit-allowed-list)
```

Rules are [strict by default but can be made more permissive with secondary options](https://stylelint.io/developer-guide/rules#write-the-rule).  By default, this rule should disallow units _anywhere in the value_, but we can add an `ignore: ["inside-function]` secondary option for your use case. Prior art for this the [`color-named`](https://stylelint.io/user-guide/rules/color-named#ignore-inside-function) rule.

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to add a new option](https://stylelint.io/developer-guide/rules#add-an-option-to-a-rule) in the Developer guide.

It should be as simple as adding the following to the rule:

https://github.com/stylelint/stylelint/blob/bcc4f9cb775c40b852051e8a7e1c8456c85f268a/lib/rules/color-named/index.js#L80-L82

And writing a couple of tests.
